TestVectorsAwsCryptographicMaterialProviders/runtimes/python/fuzz_generator.py

```diff
@@ -0,0 +1,418 @@
+        reproduced_context = {}
+        for key in draw(st.lists(st.sampled_from(context_keys), min_size=1, max_size=len(context_keys), unique=True)):
+            if draw(st.booleans()):
+                # Same key, different value (minimum size 1 to avoid empty)
+                reproduced_context[key] = draw(st.text(min_size=1, max_size=50))
+            else:
+                # Same key, same value
+                reproduced_context[key] = encryption_context[key]
+    
+    # FIXED VARIABLES: Choose algorithm suite and KMS key (randomly from valid options)
+    algorithm_suite = draw(st.sampled_from(algorithm_suites))
+    kms_key = draw(st.sampled_from(kms_keys))
+    
+    # Generate test type and keyring type independently (removing artificial restrictions)
+    test_type = draw(st.sampled_from(["positive-keyring", "negative-encrypt-keyring", "negative-decrypt-keyring"]))
+    keyring_type = draw(st.sampled_from(["kms", "raw"]))
+    
+    # Generate key descriptions based on keyring type AND test type
+    if keyring_type == "kms":
+        if test_type == "negative-encrypt-keyring":
+            # For negative encrypt tests, use configurations that will cause encryption to fail
+            # Strategy: Use required keys that don't exist in the encryption context
+            missing_required_keys = [draw(st.text(min_size=1, max_size=20)) for _ in range(draw(st.integers(min_value=1, max_value=3)))]
+            # Ensure these keys are NOT in the encryption context
+            while any(key in encryption_context for key in missing_required_keys):
+                missing_required_keys = [draw(st.text(min_size=1, max_size=20)) for _ in range(draw(st.integers(min_value=1, max_value=3)))]
+            
+            key_description = {
+                "type": "required-encryption-context-cmm",
+                "underlying": {
+                    "type": "aws-kms",
+                    "key": kms_key
+                },
+                "requiredEncryptionContextKeys": missing_required_keys  # Keys that don't exist in context
+            }
+        elif test_type == "negative-decrypt-keyring":
+            # For negative decrypt tests, use mismatched required keys between encrypt and decrypt
+            key_description = {
+                "type": "required-encryption-context-cmm",
+                "underlying": {
+                    "type": "aws-kms",
+                    "key": kms_key
+                },
+                "requiredEncryptionContextKeys": required_keys
+            }
+        else:  # positive-keyring
+            # For positive tests, use normal configuration
+            key_description = {
+                "type": "required-encryption-context-cmm",
+                "underlying": {
+                    "type": "aws-kms",
+                    "key": kms_key
+                },
+                "requiredEncryptionContextKeys": required_keys
+            }
```

Comment:
As a good coding practice, we try our best to remove duplication of code. This helps us to maintain code for a long term. I see an opportunity to reduce code duplication in line 194-201, 204-211 and 214-220. line 204-211 and 214-220 is the same code. Line 194-201 is only different by `"requiredEncryptionContextKeys"`. One idea, I have is to use use a same variable for `required_keys` and `missing_required_keys` then combine all three set of lines.